### PR TITLE
Add email notification plugin.

### PR DESCRIPTION
**Pull request by sbs2001**
Fixes https://github.com/crowdsecurity/crowdsec/issues/1003

See docs at https://github.com/crowdsecurity/crowdsec-docs/pull/96

### CODE REVIEW

File: plugins/notifications/email/email.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    {{end -}}
+  {{end -}}
+
+smtp_host: # eg value smtp.gmail.com
+smtp_username:  #Replace this with your actual username
+smtp_password:  #Replace this with your actual password
+smtp_port: # Common values are any of [25, 465, 587, 2525]
+auth_type: # Valid choices are either of "none", "crammd5", "login", "plain" 
+sender_email: # eg: foo@gmail.com
+email_subject: CrowdSec Notification
```
Comment (buixor) [inline plugins/notifications/email/email.yaml:21]: is this a go template too ?

File: plugins/notifications/email/main.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		return nil, fmt.Errorf("invalid plugin config name %s", notification.Name)
+	}
+	cfg := n.ConfigByName[notification.Name]
+	if cfg.LogLevel != nil && *cfg.LogLevel != "" {
+		logger.SetLevel(hclog.LevelFromString(*cfg.LogLevel))
+	} else {
+		logger.SetLevel(hclog.Info)
+	}
+	logger = logger.Named(cfg.Name)
+	logger.Info("got notification")
```
Comment (buixor) [inline plugins/notifications/email/main.go:73]: maybe debug ?

File: plugins/notifications/email/main.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	email.SetFrom(fmt.Sprintf("From <%s>", cfg.SenderEmail)).
+		AddTo(cfg.ReceiverEmails...).
+		SetSubject(cfg.EmailSubject)
+	email.SetBody(mail.TextHTML, notification.Text)
+
+	err = email.Send(smtpClient)
+	if err != nil {
+		return &protobufs.Empty{}, err
+	} else {
+		logger.Info(fmt.Sprintf("sent email to %v", cfg.ReceiverEmails))
```
Comment (buixor) [inline plugins/notifications/email/main.go:100]: `logger.Infof()` to be more concise

**Review by buixor — CHANGES_REQUESTED** [review]
minor changes but overall LGTM

File: plugins/notifications/email/main.go (hunk 3 above)
Comment (sbs2001) [reply plugins/notifications/email/main.go:100]: no this is hclog it doesn't have something like logger.Infof()

File: plugins/notifications/email/email.yaml (hunk 1 above)
Comment (sbs2001) [reply plugins/notifications/email/email.yaml:21]: no, the email subject would be static after the sets it in config

File: plugins/notifications/email/main.go (hunk 2 above)
Comment (sbs2001) [reply plugins/notifications/email/main.go]: done

**Review by buixor — APPROVED** [review]

**Comment by buixor** [comment]
let's merge this after 1.2.2 release

**Review by sabban — APPROVED** [review]
lgtm

**Pull request merged**